### PR TITLE
Provide API reference on readthedocs

### PR DESCRIPTION
**Pull request by hofaflo**
Sphinx is set up to build a recursive autosummary for the API reference, i.e. each new public function will be added there automatically.

A test-build is avaiable [here](https://test-sleepecg.readthedocs.io/en/latest/).

TODO:
- [x] Follow the instructions [here](https://docs.readthedocs.io/en/stable/intro/import-guide.html#importing-your-documentation) to link this repo to readthedocs. A webhook triggering builds will be created automatically. No workflow using GitHub Actions is required.
- [ ] After the next release, check whether different versions are shown and linked correctly.

### CODE REVIEW

**Comment by cbrnr** [comment]
Looks great! Minor point: it seems like we cannot use regular Markdown, because fixed-width required double backticks (see e.g. [here](https://test-sleepecg.readthedocs.io/en/latest/generated/sleepecg.io.ecg_readers.read_mitbih.html#sleepecg.io.ecg_readers.read_mitbih), the path ``<data_dir>/<db_slug>`` is not rendered correctly – I guess you wanted this to be fixed width?). Therefore, we should either check if there is a Sphinx option to allow Markdown or use the required formatting delimiters.

**Comment by hofaflo** [comment]
Ah right, thanks for noticing! Found the option and corrected all occurrences.

File: doc/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line, and also
+# from the environment for the first two.
```
Comment (cbrnr) [inline doc/Makefile:5]:
```suggestion
```

File: sleepecg/__init__.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+"""A toolbox for sleep stage classification using ECG data."""
```
Comment (cbrnr) [inline sleepecg/__init__.py:1]:
```suggestion
"""A package for sleep stage classification using ECG data."""
```

File: sleepecg/io/ecg_readers.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
 #
 # License: BSD (3-clause)
 
+"""Functions to read datasets containing ECG data and beat annotations."""
```
Comment (cbrnr) [inline sleepecg/io/ecg_readers.py:5]:
```suggestion
"""Functions for reading datasets containing ECG data and beat annotations."""
```

File: sleepecg/io/nsrr_api.py [hunk 4]
```diff
@@ -89,22 +91,23 @@ def list_nsrr_files(
...
-        Short identifier of a database, e.g. 'mesa'.
+        Short identifier of a database, e.g. `'mesa'`.
     subfolder : str, optional
-        The folder at which to start the search, by default '' (i.e. the
+        The folder at which to start the search, by default `''` (i.e. the
         root folder).
     pattern : str, optional
         Glob-like pattern to select files (only applied to the basename,
-        not the dirname!), by default '*'.
+        not the dirname!), by default `'*'`.
```
Comment (cbrnr) [inline sleepecg/io/nsrr_api.py:100]:
```suggestion
        not the dirname), by default `'*'`.
```

File: sleepecg/io/nsrr_api.py [hunk 5]
```diff
@@ -89,22 +91,23 @@ def list_nsrr_files(
...
 
     Returns
     -------
-    List[Tuple[str, str]]
-        A list of tuples (<filename>, <checksum>). <filename> is the full
-        filename (i.e. dirname and basename), <checksum> the md5-checksum.
+    list[tuple[str, str]]
+        A list of tuples `(<filename>, <checksum>)`. `<filename>` is the
+        full filename (i.e. dirname and basename), `<checksum>` the
+        md5-checksum.
```
Comment (cbrnr) [inline sleepecg/io/nsrr_api.py:110]:
```suggestion
        A list of tuples `(<filename>, <checksum>)`; `<filename>` is the
        full filename (i.e. dirname and basename) and `<checksum>` the
        MD5 checksum.
```

File: sleepecg/io/nsrr_api.py [hunk 6]
```diff
@@ -141,18 +144,18 @@ def download_nsrr_files(
...
-        Short identifier of a database, e.g. 'mesa'.
+        Short identifier of a database, e.g. `'mesa'`.
     subfolder : str, optional
-        The folder at which to start the search, by default '' (i.e. the
+        The folder at which to start the search, by default `''` (i.e. the
         root folder).
     pattern : str, optional
         Glob-like pattern to select files (only applied to the basename,
-        not the dirname!), by default '*'.
+        not the dirname!), by default `'*'`.
```
Comment (cbrnr) [inline sleepecg/io/nsrr_api.py:153]:
```suggestion
        not the dirname), by default `'*'`.
```

**Review by cbrnr — APPROVED** [review]
Besides these minor nitpicks LGTM!

**Comment by cbrnr** [comment]
Once you've applied the final changes, can you link to a rendered version of the docs?

**Comment by hofaflo** [comment]
[The previous link](https://test-sleepecg.readthedocs.io/en/latest/generated/sleepecg.html) contains the new docs now.

**Comment by cbrnr** [comment]
Looks great! What about the two bullet points in your first post? Is the first one done? Also, can you add me to the readthedocs website?

**Comment by hofaflo** [comment]
> Looks great! What about the two bullet points in your first post? Is the first one done? Also, can you add me to the readthedocs website?

The first one would have to be done by you, since I can't link the repo directly.

The second one I'll do after the next release.

**Comment by cbrnr** [comment]
Thanks @hofaflo!

**Pull request merged**